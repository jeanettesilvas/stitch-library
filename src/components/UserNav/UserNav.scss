@import '../../style/global.scss';

.user-nav {
  color: #eee;
  display: flex;
  flex-wrap: nowrap;
  font-family: $base-font;
  font-size: 14px;
  justify-content: flex-end;

  &__list {
    display: flex;
    list-style: none;
    margin: 0;
    padding: 0;
  }

  &__list-item {
    margin-left: 16px;
  }

  &__link {
    color: #333;
  }
}
