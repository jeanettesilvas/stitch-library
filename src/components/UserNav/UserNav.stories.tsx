import { Story, Meta } from '@storybook/react'

import { UserNav, UserNavProps } from './UserNav'

export default {
  title: 'Components/UserNav',
  component: UserNav,
} as Meta

const Template: Story<UserNavProps> = (args) => <UserNav {...args} />

export const Authenticated = Template.bind({})
Authenticated.args = {
  isAuthenticated: true,
  currentPathname: '/home',
}

export const NotAuthenticated = Template.bind({})
NotAuthenticated.args = {
  isAuthenticated: false,
  currentPathname: '/home',
}
