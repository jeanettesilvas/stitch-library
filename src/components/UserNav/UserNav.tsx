import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import './UserNav.scss'

export interface UserNavProps {
  dashboardUrl?: string
  hasDashboard: boolean
  isAuthenticated?: boolean
  currentPathname?: string
  profileUrl?: string
  signinUrl?: string
  signout?: () => void
}

export const UserNav = (props: UserNavProps) => {
  let {
    currentPathname,
    dashboardUrl = '/dashboard',
    hasDashboard = false,
    isAuthenticated = false,
    profileUrl = '/profile',
    signinUrl = 'sign-in',
    signout = () => {},
  } = props

  return (
    <nav className="user-nav">
      {isAuthenticated && (
        <ul className="user-nav__list">
          {hasDashboard && (
            <li className="user-nav__list-item">
              <a href={dashboardUrl} className="user-nav__link">
                <FontAwesomeIcon icon={['fal', 'home']} className="fa-lg" />
              </a>
            </li>
          )}
          <li className="user-nav__list-item">
            <a href={profileUrl} className="user-nav__link">
              <FontAwesomeIcon
                icon={['fal', 'user-circle']}
                className="fa-lg"
              />
            </a>
          </li>
          <li className="user-nav__list-item">
            <a href="/" onClick={() => signout()} className="user-nav__link">
              Sign out
            </a>
          </li>
        </ul>
      )}

      {!isAuthenticated && currentPathname !== signinUrl && (
        <a href={signinUrl} className="user-nav__link">
          Sign in
        </a>
      )}
    </nav>
  )
}
