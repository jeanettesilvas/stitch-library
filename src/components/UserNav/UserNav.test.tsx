import renderer from 'react-test-renderer'

import { UserNav } from './UserNav'

describe('UserNav', () => {
  it('renders', () => {
    const tree = renderer.create(<UserNav hasDashboard />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
