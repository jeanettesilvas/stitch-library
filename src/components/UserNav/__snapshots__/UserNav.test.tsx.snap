// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNav renders 1`] = `
<nav
  className="user-nav"
>
  <a
    className="user-nav__link"
    href="sign-in"
  >
    Sign in
  </a>
</nav>
`;
